src/components/PresenceIndicator/PresenceIndicator.tsx

```diff
@@ -0,0 +1,127 @@
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+import { Icon } from '../../'
+
+import { customPropTypes, UIComponent } from '../../lib'
+import { ComponentVariablesInput, IComponentPartStylesInput } from 'theme'
+import { Extendable } from 'utils'
+
+export interface IPresenceIndicatorProps {
+  as?: any
+  className?: string
+  size?: number
+  status?:
```

Comment:
All very Teams specific values, colours.  I think we should pass object with associates states / colours / texts that represent the possible indicators.  Then this become useful generic component.  Perhaps even presentIndicator is wrong name, perhaps stateIndicator?

Reply:
Totally agree on this. The presence (or state) indicator, as a component should provide name, color, icon and border. We should not try to define generic things here, as I am sure that in the future there will be a library that would require some additional `status` that we haven't defined. This is drawback of the Avatar component as well, and it would be great if we can extract all this in the indicator component. The project that will use the stardust, will define all different statuses they have and provide the indicators available in that project.

Reply:
Yes, we had discussed this with Levi and Brian. The plan was to first extract this in a separate component and then allow customizing the states and styles. Brian is already working on some of that.

Reply:
One way to do this is through the theme variables, like I did in the PR, but I have not seen objects in the variables so far and customizing them through the website does not work. 
Another way of doing it (as suggested by @mnajdova ) is to have props for iconName, iconColor, indicatorColor, borderColor. The consuming application can define the objects like const Available = {indicatorColor: 'green', iconName: 'check'...} etc and pass them to the component. Additionally, we could define and expose some states in the component if consumers want to use them directly. I am going to try this approach now. @kuzhelov , @hughreeling , please let me know your thoughts.

Reply:
Discussed offline with @kuzhelov , we will address the custom status issue separately. This PR will only separate out StatusIndicator in a separate component

Reply:
Updated

---
In reply to: [214710476](https://github.com/stardust-ui/react/pull/168#discussion_r214710476) [](ancestors = 214710476)